config/harvester_config.py

```diff
@@ -1,48 +1,4 @@
-import json
-import os
-
-from dotenv import load_dotenv
-
 from config.path_config import CONFIG_PATH
+from config.utils import get_harvester_config
 
-config_harvester = json.load(open(CONFIG_PATH, 'r'))
-
-try:
-    load_dotenv()
-except:
-    print('File .env not found')
-
-# Add env var secrets & pwd for swift - ovh
-config_harvester['swift'] = {}
-config_harvester['swift']['os_username'] = os.getenv('OS_USERNAME')
-config_harvester['swift']['os_password'] = os.getenv('OS_PASSWORD')
-config_harvester['swift']['os_user_domain_name'] = os.getenv('OS_USER_DOMAIN_NAME')
-config_harvester['swift']['os_project_domain_name'] = os.getenv('OS_PROJECT_DOMAIN_NAME')
-config_harvester['swift']['os_project_name'] = os.getenv('OS_PROJECT_NAME')
-config_harvester['swift']['os_project_id'] = os.getenv('OS_PROJECT_ID')
-config_harvester['swift']['os_region_name'] = os.getenv('OS_REGION_NAME')
-config_harvester['swift']['os_auth_url'] = os.getenv('OS_AUTH_URL')
-config_harvester['swift_container'] = config_harvester['publications_dump']
-
-# Add env var secrets & pwd for database - postgre
-config_harvester['db'] = {}
-config_harvester['db']['db_user'] = os.getenv('DB_USER')
-config_harvester['db']['db_password'] = os.getenv('DB_PASSWORD')
-config_harvester['db']['db_host'] = os.getenv('DB_HOST')
-config_harvester['db']['db_port'] = os.getenv('DB_PORT')
-config_harvester['db']['db_name'] = os.getenv('DB_NAME')
-
-# Add env var secrets & pwd for app - flask
-config_harvester['app'] = {}
-config_harvester['app']['flask_app'] = os.getenv('FLASK_APP')
-config_harvester['app']['flask_env'] = os.getenv('FLASK_ENV')
-config_harvester['app']['flask_ip'] = os.getenv('FLASK_IP')
-config_harvester['app']['flask_port'] = os.getenv('FLASK_PORT')
-
-# Add env var token for harvest request
-config_harvester['token'] = {}
-config_harvester['token']['wiley'] = os.getenv('TOKEN_WILEY')
-config_harvester['token']['elsevier'] = os.getenv('TOKEN_ELSEVIER')
-config_harvester['token']['springer'] = os.getenv('TOKEN_SPRINGER')
-
-
+config_harvester = get_harvester_config(CONFIG_PATH)
```

Comment:
Please, I do not well understand the need to create a new file to return the config here ? It was already the purpose of this file isn't it ?

Reply:
we are not creating a new file, we have the same file always (config.json)
I extracted the code to create the configuration dict to a function so that we can have all the configuration (general, db, wiley, ...) set for the tests which was not the case before. Before in the tests, we had only the values of the config.json in the config dict. You can check the old and new behavior in this file: "test/unit_tests/fixtures/harvester.py, l13: onfig_harvester = get_harvester_config(CONFIG_PATH_TEST)"

Reply:
we will have another ticket to refactor the configuration files

Reply:
done